.github/workflows/on_pr.yaml

```diff
@@ -48,8 +60,43 @@ jobs:
       - name: Install dependencies
         run: pip install -e .[dev]
 
-      - name: Run isort
-        run: isort .
-
-      - name: Run linter and code formatter
+      - name: Run ruff linter checks
         run: ruff check .
+
+      - name: Run ruff formatting checks
+        run: ruff format --check .
+
+  coverage:
+    runs-on: ubuntu-latest
+    needs: [tests]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal token
+          fetch-depth: 0 # otherwise, you will fail to push refs to dest repo
+
+      - name: Download coverage report
+        uses: actions/download-artifact@master
+        with:
+          name: coverageReport
+          path: tests/reports/coverage.xml
+
+      - name: Pytest coverage comment
+        if: ${{ github.ref == 'refs/heads/main' }}
+        id: coverageComment
+        uses: MishaKav/pytest-coverage-comment@main
+        with:
+          hide-comment: true
+          pytest-xml-coverage-path: ./tests/reports/coverage.xml
+
+      - name: Update Readme with Coverage Html
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: |
+          sed -i '/<!-- Pytest Coverage Comment:Begin -->/,/<!-- Pytest Coverage Comment:End -->/c\<!-- Pytest Coverage Comment:Begin -->\n\${{ steps.coverageComment.outputs.coverageHtml }}\n<!-- Pytest Coverage Comment:End -->' ./README.md
+
+      - name: Commit & Push changes to Readme
+        if: ${{ github.ref == 'refs/heads/main' }}
+        uses: actions-js/push@master
+        with:
+          message: Update coverage on Readme
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
To reduce the 3rd party dependencies, and instead of using an unknown github action where literally `git add .., git commit, git push` commands run.
It could be even more dangerous, because there could be malicious code, especially when we pass `GITHUB_TOKEN` with so many permissions: `content, checks,pull-requests` to the untrusted 3rd party action, like `actions-js/push`.

I'm suggesting to write it explicitly in this case:
```

    - name: Git commit
      id: commit
      run: |
        git config --global user.name 'GitHub Actions Workflow glassflow-python-sdk/.github/workflows/on_pr.yaml'
        git config --global user.email 'glassflow-actions-workflow@users.noreply.github.com'

        git add README.md
        git commit -m "Update coverage on Readme"

        git push origin master
```

Reply:
Done ✅

Reply:
Thank you!🚀